graphql/distro_resolver.go

```diff
@@ -48,6 +48,26 @@ func (r *distroResolver) CloneMethod(ctx context.Context, obj *model.APIDistro)
 	}
 }
 
+// Provider is the resolver for the provider field.
+func (r *distroResolver) Provider(ctx context.Context, obj *model.APIDistro) (Provider, error) {
+	if obj == nil {
+		return "", InternalServerError.Send(ctx, "distro undefined when attempting to resolve provider")
+	}
+
+	switch utility.FromStringPtr(obj.Provider) {
+	case evergreen.ProviderNameDocker:
+		return ProviderDocker, nil
+	case evergreen.ProviderNameEc2Fleet:
+		return ProviderEc2Fleet, nil
+	case evergreen.ProviderNameEc2OnDemand:
+		return ProviderEc2Ondemand, nil
```

Comment:
nit: can this be OnDemand? or is this generated in such a way that we'd then have to do ON_DEMAND for the enum? I might still prefer that